### PR TITLE
User-follows

### PR DESCRIPTION
**Pull request by lindeberg**
User follows shows up!
More work to be done here though.

### CODE REVIEW

File: Snitch.Core/Extensions/StringExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MoreLinq;
+
+namespace Snitch.Core.Extensions
+{
+    public static class StringExtensions
+    {
+        public static string ToQueryString(this IEnumerable<string> ids, string parameterName = "id")
```
Comment (Zexuz) [inline Snitch.Core/Extensions/StringExtensions.cs]: A more generic way?
```csharp
  public static string GetQueryString(Dictionary<string,object> parameters)
        {
            var queryString = HttpUtility.UrlEncode(string.Join("&", parameters.Select(kvp => $"{kvp.Key}={kvp.Value}")));
            return queryString;
        }
```

File: Snitch.Core/Services/IServiceProviderWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Snitch.Core.Services
+{
+    public interface IServiceProviderWrapper
+    {
+        T Resolve<T>();
```
Comment (Zexuz) [inline Snitch.Core/Services/IServiceProviderWrapper.cs]: Maybe pick one? Only resolve method or have all servicies as properties, bot not both?

File: Snitch.Core/TwitchApi/TwitchHelixApiClient.cs [hunk 3]
```diff
@@ -18,6 +21,34 @@ public TwitchHelixApiClient(HelixHttpCallerService callerService)
             var url = $"/users?description={WebUtility.UrlEncode(description)}";
             return await _callerService.Put(url);
         }
-        
+
```
Comment (Zexuz) [inline Snitch.Core/TwitchApi/TwitchHelixApiClient.cs]: Use the new QueryString extension method to generate the new QueryStrings?

File: Snitch.Core/Services/UserService.cs [hunk 4]
```diff
@@ -26,18 +28,95 @@ IMapper mapper
             if (res.IsSuccessStatusCode)
                 return await res.Content.ReadAsStringAsync();
 
-            throw new Exception("NOne successful status code returned");
+            throw new Exception("None successful status code returned");
+        }
+
+        public async Task<UserFollowsViewModel> GetUserFollowsViewModel()
```
Comment (lindeberg) [inline Snitch.Core/Services/UserService.cs]: @Zexuz Should a service return ViewModel?

File: Snitch.Core/Extensions/StringExtensions.cs (hunk 1 above)
Comment (lindeberg) [reply Snitch.Core/Extensions/StringExtensions.cs]: `public static string ToQueryString(this IEnumerable<KeyValuePair<string, string>> parameters)
        {
            return HttpUtility.UrlEncode("?" + string.Join("&", parameters.Select(kvp => $"{kvp.Key}={kvp.Value}")));
        }`

` public async Task<HttpResponseMessage> GetUsers(IEnumerable<string> followUserIds = null)
        {
            var queryString = followUserIds?
                .Select(x => new KeyValuePair<string, string>("id", x))
                .ToQueryString();

            return await _callerService.Get($"users{queryString}");
        }`

File: Snitch.Core/Services/IServiceProviderWrapper.cs (hunk 2 above)
Comment (lindeberg) [reply Snitch.Core/Services/IServiceProviderWrapper.cs]: Private Resolve<T>(); ?

Comment (Zexuz) [reply Snitch.Core/Services/IServiceProviderWrapper.cs]: Yes

File: Snitch.Core/Extensions/Extensions.cs [hunk 5]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Web;
+
+namespace Snitch.Core.Extensions
+{
+    public static class Extensions
+    {
+        public static string ToQueryString(this IEnumerable<KeyValuePair<string, string>> parameters)
```
Comment (Zexuz) [inline Snitch.Core/Extensions/Extensions.cs]: Nitpicking now, but if it's a string, string. How do you pass int to it? 1.ToString()?
Would it not be better to have it, string,object and then call obj.ToString();

**Pull request merged**